mesosphere/kudo-spark-operator
# [DCOS-59168] grafana dashboard

The PR addresses Spark Operator and Application monitoring dashboards. Which based on metrics scraped by Prometheus from Spark Operator, Driver and Executors, ran in Kubernetes cluster.

### How to verify
1) Spin up Konvoy cluster
2) Follow instruction in [README.md](https://github.com/mesosphere/kudo-spark-operator/tree/DCOS-59168_Grafana_dashboard#submitting-spark-application) to deploy Spark Operator and application example.
3) Go to Grafana and press + then select Import
4) Copy the dashboard code from [grafana_spark_operator](https://github.com/mesosphere/kudo-spark-operator/blob/DCOS-59168_Grafana_dashboard/grafana/grafana_spark_operator.json) and paste it to text area in Grafana
5) Save and select the "Spark Operator" dashboard to play around.

And repeat steps 3 - 5 for  [spark_application_dashboard.json](https://github.com/mesosphere/kudo-spark-operator/blob/DCOS-59168_Grafana_dashboard/grafana/spark_application_dashboard.json)  if you want check the Spark Driver and Executors metrics dashboard.

Find a couple of screenshots here Spark Driver and Executors metrics:
<img width="1661" alt="Screen Shot 2019-10-01 at 02 13 05" src="https://user-images.githubusercontent.com/7896911/65923606-34446880-e3f2-11e9-9838-4a1de5842a25.png">
<img width="1661" alt="Screen Shot 2019-10-01 at 02 14 29" src="https://user-images.githubusercontent.com/7896911/65923607-34446880-e3f2-11e9-8f13-2c669e99d65a.png">
<img width="1661" alt="Screen Shot 2019-10-01 at 02 14 55" src="https://user-images.githubusercontent.com/7896911/65923608-34446880-e3f2-11e9-874f-070dc69c7303.png">
<img width="1661" alt="Screen Shot 2019-10-01 at 02 15 15" src="https://user-images.githubusercontent.com/7896911/65923609-34dcff00-e3f2-11e9-9bc8-64ce9d334173.png">
<img width="1661" alt="Screen Shot 2019-10-01 at 02 15 38" src="https://user-images.githubusercontent.com/7896911/65923610-34dcff00-e3f2-11e9-853a-2fc51456d994.png">
<img width="1661" alt="Screen Shot 2019-10-01 at 02 15 53" src="https://user-images.githubusercontent.com/7896911/65923611-34dcff00-e3f2-11e9-8c7a-9831d3401c76.png">
<img width="1661" alt="Screen Shot 2019-10-01 at 02 16 24" src="https://user-images.githubusercontent.com/7896911/65923612-34dcff00-e3f2-11e9-8687-22687f6d437a.png">
<img width="1661" alt="Screen Shot 2019-10-01 at 02 16 53" src="https://user-images.githubusercontent.com/7896911/65923613-34dcff00-e3f2-11e9-9e28-62ea62228506.png">
